test/test_torch.py

```diff
@@ -187,10 +200,14 @@ def test_horovod_allreduce_async_fused(self):
         with Tensor Fusion."""
         hvd.init()
         size = hvd.size()
-        dtypes = [torch.IntTensor, torch.LongTensor,
-                  torch.FloatTensor, torch.DoubleTensor]
-        if _fp16_supported:
-            dtypes += [torch.HalfTensor]
+        # MLSL supports only byte, float and double data types
+        if 'MLSL_ROOT' in os.environ:
```

Comment:
Looks like almost every test has been updated with this check.  I think we can simplify this and make it more extensible to other frameworks with data type restrictions by having a single utility function that filters the list of dtypes down to only the ones supported by the current framework.

For example:

```
MLSL_TYPES = set([torch.FloatTensor, torch.DoubleTensor])

def filter_supported_types(dtypes):
    if 'MLSL_ROOT' in os.environ:
        dtypes = [dtype for dtype in dtypes if dtype in MLSL_TYPES]
    return dtypes

...
dtypes = filter_supported_types([torch.IntTensor, torch.LongTensor,
                                                         torch.FloatTensor, torch.DoubleTensor])
```

Reply:
Thank you. Done